core/dbt/contracts/util.py

```diff
@@ -234,9 +255,11 @@ def read_and_check_versions(cls, path: str):
                 # cls.dbt_schema_version is a SchemaVersion object
                 if not cls.is_compatible_version(previous_schema_version):
                     raise IncompatibleSchemaException(
-                        expected=str(cls.dbt_schema_version), found=previous_schema_version
+                        expected=str(cls.dbt_schema_version),
+                        found=previous_schema_version,
                     )
-
+        if get_manifest_schema_version(data) <= 6:
```

Comment:
Logic to load existing manifest pre update, double check version before merge/release